src/routes/github/github-oauth-router.ts

```diff
@@ -205,8 +206,45 @@ const getCloudOrGHESAppClientSecret = async (gitHubAppConfig, jiraHost: string)
 	return ghesApp.getDecryptedGitHubClientSecret(jiraHost);
 };
 
+const TempGetJiraHostFromStateMiddleware  = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
At the moment, even for cloud, `jiraHost` is missing on oauth callback.
This is okay for cloud as cloud doesn't need it during callback.
But for GHE it needs the jiraHost to decrypt the gitHubApp to get secrets.

I am using req.session[state] as this supposed to be securely set during oauth get.